File: CS/DXWebApplication1/Models/DBHelper.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace DXWebApplication1.Models {

    public class DBHelper {

        const string NorthwindDataContextKey = "DXNorthwindDataContext";

        public static NwindDataClassesDataContext DB {
            get {
                if (HttpContext.Current.Items[NorthwindDataContextKey] == null)
                    HttpContext.Current.Items[NorthwindDataContextKey] = new NwindDataClassesDataContext();
                return (NwindDataClassesDataContext)HttpContext.Current.Items[NorthwindDataContextKey];
            }
        }

        public static int GetKeyIdByDisplayIndex(int? displayIndex) {
            var model = DB.Products;
            var rowKey = model.Where(q => q.DisplayIndex == displayIndex).Select(q => q.ProductID).FirstOrDefault();
            return rowKey;
        }

        public static void UpdateDisplayIndex(int? rowKey, int? displayIndex) {
            var model = DB.Products;
            var product = model.Where(q => q.ProductID == rowKey).FirstOrDefault();
            product.DisplayIndex = displayIndex;
            DB.SubmitChanges();
        }
    }
}

File: VB/DXWebApplication1/Models/DBHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web

Namespace DXWebApplication1.Models

	Public Class DBHelper

		Private Const NorthwindDataContextKey As String = "DXNorthwindDataContext"

		Public Shared ReadOnly Property DB() As NwindDataClassesDataContext
			Get
				If HttpContext.Current.Items(NorthwindDataContextKey) Is Nothing Then
					HttpContext.Current.Items(NorthwindDataContextKey) = New NwindDataClassesDataContext()
				End If
				Return CType(HttpContext.Current.Items(NorthwindDataContextKey), NwindDataClassesDataContext)
			End Get
		End Property

		Public Shared Function GetKeyIdByDisplayIndex(ByVal displayIndex As Nullable(Of Integer)) As Integer
			Dim model = DB.Products
			Dim rowKey = model.Where(Function(q) q.DisplayIndex.Equals(displayIndex)).Select(Function(q) q.ProductID).FirstOrDefault()
			Return rowKey
		End Function

		Public Shared Sub UpdateDisplayIndex(ByVal rowKey As Nullable(Of Integer), ByVal displayIndex As Nullable(Of Integer))
			Dim model = DB.Products
			Dim product = model.Where(Function(q) q.ProductID.Equals(rowKey)).FirstOrDefault()
			product.DisplayIndex = displayIndex
			DB.SubmitChanges()
		End Sub
	End Class
End Namespace